src/runtime_src/xdp/profile/device/pl_device_intf.cpp

```diff
@@ -604,28 +602,28 @@ void PLDeviceIntf::readDebugIPlayout() {
       for (uint64_t i = 0; i < map->m_count; i++) {
         switch (map->m_debug_ip_data[i].m_type) {
         case AXI_MM_MONITOR:
-          mAimList.push_back(new AIM(mDevice, i, &(map->m_debug_ip_data[i])));
+          mAimList.push_back(new AIM(mDevice.get(), i, &(map->m_debug_ip_data[i])));
```

Comment:
Maybe these can also be converted into smart pointers using std::make_unique in future